src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,271 @@
+        return False
+
+    selected = [sel.Object for sel in selx if bool(sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::Face") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
+    if len(selected) != 1:
+        App.Console.PrintLog(f"makeRevolve() only 1 selected profile needed, not skipping dialog\n")
+        return False
+
+    solid = "True"
+    base = selected[0]
+    if hasattr(base.Shape, "Wire1"):
+        if not base.Shape.Wire1.isClosed():
+            solid = "False"
+    else:
+        if not bool(hasattr(base,"Shape") and len(base.Shape.Vertexes) == 1):
+            App.Console.PrintLog(f"makeRevolve() selected object has no wire and not a vertex, not skipping dialog\n")
+            return False
+        else:
+            solid = "False"
+
+    spines = [sel for sel in selx if sel.HasSubObjects]
+    if len(spines) != 1:
+        App.Console.PrintLog(f"makeRevolve() all edges must be from the same object, not skipping dialog\n")
+        return False
+
+    spineEdges = spines[0].SubElementNames
+    subNames = [name for name in spineEdges if "Edge" in name]
+    if len(subNames) != 1:
+        App.Console.PrintLog(f"makeRevolve() only 1 edge may be selected, not skipping dialog\n")
+        return False
+
+    if solid == "False":
+        App.Console.PrintWarning("Solid = False\n")
+
+    doc = selected[0].Document
+    doc.openTransaction("Revolve")
+    cmd = getAddObjStr(doc, "Part::Revolution", "Revolve")
+    cmd += getActStr(doc) + ".Source = " + getObjStr(doc, selected[0]) + "\n"
+    cmd += getActStr(doc) + ".Solid = " + solid + "\n"
+    cmd += getActStr(doc) + ".Symmetric = False\n"
+    cmd += getActStr(doc) + ".Angle = 360.0\n"
+    cmd += setHiddenStr(doc, selected)
+    cmd += getActStr(doc) + ".AxisLink = (" + getObjStr(doc, spines[0].Object) + ", \'" + subNames[0] + "\')\n"
+    Gui.doCommand(cmd)
+    doc.commitTransaction()
+    doc.recompute()
+    return True
+
+
+def makeLoft():
+
+    selx = Gui.Selection.getSelectionEx()
+    if len(selx) == 0:
+        App.Console.PrintLog(f"makeLoft() nothing selected, not skipping dialog\n")
+        return False
+
+    selected = [sel.Object for sel in selx if sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::Face") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1)]
+    if len(selected) < 2:
+        App.Console.PrintLog(f"makeLoft() need at least 2 objects selected, not skipping dialog\n")
+        return False
+
+    #filter out sketches with multiple wires
```

Comment:
Does Part/Extrude and Part/Revolve support sketch with multiple wire ?

Reply:
They can even manage this monster with all the nested wires.
![Snip macro screenshot-5d1441](https://user-images.githubusercontent.com/39143564/154326202-6025f539-fcac-4072-bc81-c171134951f8.png)